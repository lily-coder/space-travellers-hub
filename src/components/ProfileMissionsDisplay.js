import React from 'react';
import { useSelector } from 'react-redux';

const ProfileMissionsDisplay = () => {
  const data = useSelector((state) => state.missions);

  return (
    <div>
      <ul>
        {data.filter((mission) => mission.joined).map((mission) => (
          <div key={mission.mission_id}>
            <li className='profile-rocket-name'>{mission.mission_name}</li>
          </div>
        ))}
      </ul>
    </div>
  );
};

export default ProfileMissionsDisplay;
