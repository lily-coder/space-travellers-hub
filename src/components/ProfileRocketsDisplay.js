/* eslint-disable no-unused-vars */
import React from 'react';
import { useSelector } from 'react-redux';

const ProfileRocketsDisplay = () => {
  const myRocketArray = useSelector((state) => state.rockets);

  return (
    <div>
      <ul>
        {myRocketArray.filter((rocket) => rocket.reserved).map((rocket) => (
          <div key={rocket.id}>
            <li className='profile-rocket-name'>{rocket.rocket_name}</li>
          </div>
        ))}
      </ul>
    </div>
  );
};

export default ProfileRocketsDisplay;
