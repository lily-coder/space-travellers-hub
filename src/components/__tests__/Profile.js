import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import { Provider } from 'react-redux';
import store from '../../redux/configStore.js';
import ProfileMissionsDisplay from '../ProfileMissionsDisplay.js';
import ProfileRocketsDisplay from '../ProfileRocketsDisplay.js';

describe('Render the rockets and profile list', () => {
  it('Renders the rockets and profile list correctly', () => {
    const rockets = TestRenderer.create(
        <Provider store={store}>
          <ProfileMissionsDisplay />
          <ProfileRocketsDisplay />
        </Provider>,
    ).toJSON();
    expect(rockets).toMatchSnapshot();
  });
});