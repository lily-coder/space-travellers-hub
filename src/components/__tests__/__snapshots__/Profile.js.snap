// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the rockets and profile list Renders the rockets and profile list correctly 1`] = `
Array [
  <div>
    <ul />
  </div>,
  <div>
    <ul />
  </div>,
]
`;
